### PR TITLE
Adds a lot of components

### PR DESCRIPTION
**Pull request by andrewkfiedler**
- Take note of how the wrapping of component is done to allow us to style them (with styled).

### CODE REVIEW

File: atoms/dropdown/dropdown.context.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+   * Close the dropdown and refocus on the element that opened it
+   */
+  deepCloseAndRefocus: () => void
+  /**
+   * Close the dropdown and refocus on the element that opened it
+   */
+  depthCloseAndRefocus: (depth: number) => void
+}
+
+export const DropdownContext = React.createContext({
```
Comment (Bdthomson) [inline atoms/dropdown/dropdown.context.tsx]: ✏️ `useContext()`?

File: atoms/grid/grid.tsx [hunk 2]
```diff
@@ -1,3 +1,55 @@
...
-export default Grid
+export const WrappedGrid = styled(
+  React.forwardRef((props: GridProps, ref: React.Ref<any>) => {
+    return <Grid {...props} ref={ref} />
+  })
+)<GridProps>``
+
+const GridItem = styled(WrappedGrid)`
+  > * {
+    height: 100%;
```
Comment (Bdthomson) [inline atoms/grid/grid.tsx]: Do we want to be hardcoding this?

File: atoms/input/autocomplete.tsx [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+import { emphasize, makeStyles, useTheme } from '@material-ui/core/styles'
+import Typography from '@material-ui/core/Typography'
+import TextField from '@material-ui/core/TextField'
+import Paper from '@material-ui/core/Paper'
+import Chip from '@material-ui/core/Chip'
+import MenuItem from '@material-ui/core/MenuItem'
+import CancelIcon from '@material-ui/icons/Cancel'
+import { Props as CreatableProps } from 'react-select/src/Creatable'
+
+const useStyles = makeStyles(theme => ({
```
Comment (Bdthomson) [inline atoms/input/autocomplete.tsx]: Is there a reason you are using makeStyles rather than `styled`?

File: typescript/index.tsx [hunk 4]
```diff
@@ -1,2 +1,3 @@
 export { default as Omit } from './omit'
 export { default as Subtract } from './subtract'
+export { setType } from './hooks'
```
Comment (Bdthomson) [inline typescript/index.tsx]: Hooks should follow the `useX` naming convention right?

Comment (andrewkfiedler) [reply typescript/index.tsx]: This is a convenience type generator for useState.

Here's some example usage.
```
const ChildComponent = ({setState}: {setState: setType<string>}) => {
  ..... whatever
}

const ParentComponent = () => {
   const [state, setState] = React.useState("")
   return <ChildComponent setState={setState} />
}
```

File: atoms/input/autocomplete.tsx (hunk 3 above)
Comment (andrewkfiedler) [reply atoms/input/autocomplete.tsx]: Not sure if we have to or not, I copied most of this code from https://github.com/mui-org/material-ui/blob/master/docs/src/pages/components/autocomplete/IntegrationReactSelect.js .  I'm guessing we could pare it down a bit in the future.

File: atoms/grid/grid.tsx (hunk 2 above)
Comment (andrewkfiedler) [reply atoms/grid/grid.tsx]: I think so, it's only for the WrappedCardGrid component.

**Pull request merged**